src/main/java/org/gridsuite/shortcircuit/server/entities/AnalysisParametersEntity.java

```diff
@@ -0,0 +1,137 @@
+    @Enumerated(EnumType.STRING)
+    private ShortCircuitPredefinedConfiguration predefinedParameters = ShortCircuitPredefinedConfiguration.ICC_MAX_WITH_NOMINAL_VOLTAGE_MAP;
+
+    @Builder.Default
+    @Column(name = "withLoads", columnDefinition = "boolean default false")
+    private boolean withLoads = false;
+
+    @Builder.Default
+    @Column(name = "withShuntCompensators", columnDefinition = "boolean default false")
+    private boolean withShuntCompensators = false;
+
+    @Builder.Default
+    @Column(name = "withVscConverterStations", columnDefinition = "boolean default true")
+    private boolean withVscConverterStations = true;
+
+    @Builder.Default
+    @Column(name = "withNeutralPosition", columnDefinition = "boolean default true")
+    private boolean withNeutralPosition = true;
+
+    @Builder.Default
+    @Column(name = "initialVoltageProfileMode", columnDefinition = "varchar(255) default 'NOMINAL'")
+    @Enumerated(EnumType.STRING)
+    private InitialVoltageProfileMode initialVoltageProfileMode = InitialVoltageProfileMode.NOMINAL;
+
+    public AnalysisParametersEntity updateWith(final AnalysisParametersEntity source) {
+        setIfDifferent(this, source, AnalysisParametersEntity::isWithLimitViolations, AnalysisParametersEntity::setWithLimitViolations);
+        setIfDifferent(this, source, AnalysisParametersEntity::isWithVoltageResult, AnalysisParametersEntity::setWithVoltageResult);
+        setIfDifferent(this, source, AnalysisParametersEntity::isWithFortescueResult, AnalysisParametersEntity::setWithFortescueResult);
+        setIfDifferent(this, source, AnalysisParametersEntity::isWithFeederResult, AnalysisParametersEntity::setWithFeederResult);
+        setIfDifferent(this, source, AnalysisParametersEntity::getStudyType, AnalysisParametersEntity::setStudyType);
+        setIfDifferent(this, source, AnalysisParametersEntity::getMinVoltageDropProportionalThreshold, AnalysisParametersEntity::setMinVoltageDropProportionalThreshold);
+        setIfDifferent(this, source, AnalysisParametersEntity::getPredefinedParameters, AnalysisParametersEntity::setPredefinedParameters);
+        setIfDifferent(this, source, AnalysisParametersEntity::isWithLoads, AnalysisParametersEntity::setWithLoads);
+        setIfDifferent(this, source, AnalysisParametersEntity::isWithShuntCompensators, AnalysisParametersEntity::setWithShuntCompensators);
+        setIfDifferent(this, source, AnalysisParametersEntity::isWithVscConverterStations, AnalysisParametersEntity::setWithVscConverterStations);
+        setIfDifferent(this, source, AnalysisParametersEntity::isWithNeutralPosition, AnalysisParametersEntity::setWithNeutralPosition);
+        setIfDifferent(this, source, AnalysisParametersEntity::getInitialVoltageProfileMode, AnalysisParametersEntity::setInitialVoltageProfileMode);
+        return this;
+    }
+
+    private static <T> void setIfDifferent(@NotNull final AnalysisParametersEntity dst, @NotNull final AnalysisParametersEntity src,
```

Comment:
If a parameter is not different it does not change anything so I feel like this check is not necessary. You should rather check if the value is null, no ? If it's null it means you shouldn't override and itf it is present you should override

Reply:
I'm doing this for
* having only one get (optimization and help in tests)
* helping hibernate to know if the entity have been updated, because attached entities are in fact proxies who trace when setters have been called to know it must be saved at the end of the transaction or not

Reply:
Keep it simple first we'll do optimizations when we would need optimizations. I'm not convinced this one is necessary